Snippets/Core/Core_3/CriticalErrors.cs

```diff
@@ -26,16 +26,20 @@ class CriticalErrors
 
         void OnCriticalError()
         {
-            // Write log entry in version 3 since this is not done by default.
-            log.Fatal("CRITICAL Error");
-
-            // To leave the process active, dispose the bus.
-            // When the bus is disposed sending messages will throw an ObjectDisposedException.
-            ((IDisposable)bus).Dispose();
-
-            // To kill the process, raise a fail fast error as shown below.
-            // var failMessage = "Critical error shutting.";
-            // Environment.FailFast(failMessage);
+            try
+            {
+                // To leave the process active, dispose the bus.
+                // When the bus is disposed, the attempt to send message will cause an ObjectDisposedException.
+                ((IDisposable)bus).Dispose();
+                // Perform custom actions here, e.g.
+                // NLog.LogManager.Shutdown();
+            }
+            finally
+            {
+                // Write log entry in version 3 since this is not done by default.
+                log.Fatal("CRITICAL Error");
```

Comment:
If logging, that should be done within the `try` as that call could fail and then `FailFast` will not be invoked.